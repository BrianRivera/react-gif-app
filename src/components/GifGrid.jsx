import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
// import { getGif } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {

// const [images, setImages] = useState([])

const {data:images,loading} = useFetchGifs(category);

    return (
        <>
        <h3 className="animate__animated animate__bounce">{category}</h3>
        {loading && <p className="animate__animated animate__flash">Loading...</p>}
        <div className="card-grid">
                {
                    images.map((resp) =>(
                        <GifGridItem 
                        key={resp.id}
                        img={resp}
                        ></GifGridItem>
                    ))
                }
        </div>
        </>
    )
}


/*
<GifGridItem 
key={img.id}
{...img}
></GifGridItem>
devuelve todas las propiedades contenidos en el objeto
como un prop individual
*/
