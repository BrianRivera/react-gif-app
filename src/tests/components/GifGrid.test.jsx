import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('GifGrid test', () => {
    const category = 'sasha grey';
    
    test('componente hace match', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}></GifGrid>);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        //un mock es para finjir que el componente tiene algo
        const gifs = [{
            id:'ABC',
            url: 'http://cosa/cosa',
            title: 'gif prueba'
        },
        {
            id:'ABC',
            url: 'http://cosa/cosa',
            title: 'gif prueba'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}></GifGrid>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

    });
    


})
