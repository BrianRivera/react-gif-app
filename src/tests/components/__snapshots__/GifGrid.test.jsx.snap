// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid test componente hace match 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    sasha grey
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid test debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    sasha grey
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="gif prueba"
      url="http://cosa/cosa"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="gif prueba"
      url="http://cosa/cosa"
    />
  </div>
</Fragment>
`;
